packages/react-refresh/src/ReactFreshRuntime.js

```diff
@@ -195,8 +195,17 @@ export function performReactRefresh(): RefreshUpdate | null {
 
     let didError = false;
     let firstError = null;
-    failedRoots.forEach((element, root) => {
-      const helpers = helpersByRoot.get(root);
+
+    // We snapshot maps and sets that are mutated during commits.
+    // If we don't do this, there is a risk they will be mutated while
+    // we iterate over them. For example, trying to recover a failed root
+    // may cause another root to be added to the failed list -- an infinite loop.
+    let failedRootsSnapshot = new Map(failedRoots);
```

Comment:
is it supposed that `new Map([iterable])` is polyfilled in `react-refresh`

Reply:
Would IE11 be the only one that can’t do it? I guess we could use something longer but tbh I don’t know if we care about supporting IE11 for a developer productivity feature like this.